### PR TITLE
Add edit task functionality with view.

### PR DESCRIPTION
**Pull request by jhryniuk**
@partikus @tarnawski @dominikhajduk CR please.

### CODE REVIEW

**Comment by tarnawski** [comment]
Can one of the admins verify this patch?

File: spec/TaskManager/Repository/TaskRepositorySpec.php [hunk 1]
```diff
@@ -21,7 +21,7 @@ function it_is_initializable()
 
     function it_should_return_all_task(TaskStorage $storage)
     {
-        $data = [[
+        $data = [1 => [
```
Comment (dominikhajduk) [inline spec/TaskManager/Repository/TaskRepositorySpec.php]: is that required when id is unused?

File: spec/TaskManager/Repository/TaskRepositorySpec.php [hunk 2]
```diff
@@ -31,7 +31,7 @@ function it_should_return_all_task(TaskStorage $storage)
         $this->beConstructedWith($storage);
 
         $result = $this->getAll();
-        $first = $result[0];
+        $first = $result['1'];
```
Comment (dominikhajduk) [inline spec/TaskManager/Repository/TaskRepositorySpec.php]: Maybe better array_pop / array_shift

File: src/TaskManager/Controller/TaskController.php [hunk 3]
```diff
@@ -61,4 +61,27 @@ public function createAction()
...
         }
     }
+
+    public function editAction($params)
+    {
+        /** @var Storage $storage */
+        $storage = $this->get('task_storage_in_xml');
+        $taskRepository = new TaskRepository($storage);
+        $task = $taskRepository->getSingle($params[0]);
+        if (!isset($_POST['submit'])) {
```
Comment (dominikhajduk) [inline src/TaskManager/Controller/TaskController.php]: Else is not needed.
```php
if (!isset($_POST['submit'])) {
    return $this->render('task/edit.html.twig', ['task' => $task]);
}

$name = isset($_POST['name']) ? $_POST['name'] : '';
// ... rest of stuff..
```

File: src/TaskManager/View/task/edit.html.twig [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{% extends "base.html.twig" %}
+
+{% block content %}
+    <h2>
+        EDIT TASK {{ task.id }}
+        <a href="/tasks" class="btn btn-primary pull-right" role="button">BACK TO LIST</a>
+    </h2>
+    <br>
```
Comment (dominikhajduk) [inline src/TaskManager/View/task/edit.html.twig]: `<br>` or `<br />`

File: spec/TaskManager/Repository/TaskRepositorySpec.php (hunk 1 above)
Comment (jhryniuk) [inline spec/TaskManager/Repository/TaskRepositorySpec.php]: in order to better understanding, please take look at this
`https://github.com/tarnawski/task-manager/pull/23/files#diff-6967a92ef83c357144cd40a4d817007fR25`

File: spec/TaskManager/Repository/TaskRepositorySpec.php (hunk 2 above)
Comment (jhryniuk) [inline spec/TaskManager/Repository/TaskRepositorySpec.php]: After using those `array` functions $result going to be empty array. I do not need that.

**Comment by tarnawski** [comment]
After changes looks good for me 👍

**Comment by tarnawski** [comment]
ok to test

**Pull request merged**